dashboard/src/components/AppList/AppList.test.tsx

```diff
@@ -1,36 +1,28 @@
 import { shallow } from "enzyme";
+import context from "jest-context";
 import * as React from "react";
 
-import sharedSpecs from "../../shared/specs";
+import itBehavesLike from "../../shared/specs";
 import { IAppOverview, IAppState } from "../../shared/types";
 import { CardGrid } from "../Card";
 import AppList from "./AppList";
 import AppListItem from "./AppListItem";
 
-let defaultProps = {} as any;
 let props = {} as any;
 
-beforeEach(() => {
-  defaultProps = {
-    apps: {} as IAppState,
-    fetchApps: jest.fn(),
-    filter: "",
-    namespace: "default",
-    pushSearchFilter: jest.fn(),
-    toggleListAll: jest.fn(),
-  };
-});
+const defaultProps: any = {
+  apps: {} as IAppState,
+  fetchApps: jest.fn(),
+  filter: "",
+  namespace: "default",
+  pushSearchFilter: jest.fn(),
+  toggleListAll: jest.fn(),
+};
 
-describe("while fetching apps", () => {
-  beforeEach(() => {
-    props = { ...defaultProps, apps: { isFetching: true } };
-  });
+context("while fetching apps", () => {
```

Comment:
what's the difference between context and describe?

Reply:
It is just syntax sugar. I like using describe to mention the test subject and context to set scenarios.

It might now be worth adding the library that's why I did not load it globally, let me know if you'd prefer not to have it.

Reply:
okay, was just wondering if there was actually some difference, thanks!

Reply:
Good catch @andresmgot.

  >  Also, in my opinion, we should try to use avoid using third-party modules if possible (to avoid possible vulnerabilities or license issues).

This is tricky, because this single module is nothing compared to all the dependencies from our main components.

I do not see any license issue in this case since it is MIT.

>  In any case I don't think we should add it, as happened with @prydonius or myself, other people may be confused with the context usage.

What do you find confusing? Having two methods? This concept [comes from rspec](http://www.betterspecs.org/#contexts) in which describe and context are used for different things. [Context used to setup a context or scenario and describe to express the test subject](http://lmws.net/describe-vs-context-in-rspec).

I can remove it and use nested describes but honestly that looks odd and adding `context` helps with tests readability (at least for me). If you are strongly opinionated against it I can remove it.

Reply:
It's a minor thing, I know that in this case it's just a minor package with a single line. I am not against it so go ahead if you want to use it. I guess I am more used to see nested `describe`s so I don't find it weird.